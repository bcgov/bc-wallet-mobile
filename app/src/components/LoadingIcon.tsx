import React, { useEffect, useRef } from 'react'
import { Animated } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

interface LoadingIconProps {
  size: number
  color: string
  active: boolean
}

const LoadingIcon: React.FC<LoadingIconProps> = ({ size, color, active }) => {
  const rotationAnim = useRef(new Animated.Value(0)).current
  const timing: Animated.TimingAnimationConfig = {
    toValue: 1,
    duration: 2000,
    useNativeDriver: true,
  }
  const rotation = rotationAnim.interpolate({
    inputRange: [0, 1],
    outputRange: ['0deg', '360deg'],
  })

  const animation = Animated.loop(Animated.timing(rotationAnim, timing))

  useEffect(() => {
    animation.reset()
    if (active) {
      animation.start()
    } else {
      animation.stop()
    }
  }, [rotationAnim, active])

  return (
    <Animated.View style={[{ transform: [{ rotate: rotation }] }]}>
      <Icon style={{ color }} size={size} name="refresh" />
    </Animated.View>
  )
}

export default LoadingIcon
