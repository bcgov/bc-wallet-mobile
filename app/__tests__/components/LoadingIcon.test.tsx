import { render } from '@testing-library/react-native'
import React from 'react'

import LoadingIcon from '../../src/components/LoadingIcon'

describe('LoadingIcon Component', () => {
  test('renders correctly', () => {
    const size = 50 // arbitrary
    const color = '#333' // arbitrary
    const active = true // arbitrary
    const tree = render(<LoadingIcon size={size} color={color} active={active} />)
    expect(tree).toMatchSnapshot()
  })
})
