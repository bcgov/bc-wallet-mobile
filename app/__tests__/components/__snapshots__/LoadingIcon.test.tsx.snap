// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIcon Component renders correctly 1`] = `
<View
  collapsable={false}
  nativeID="animatedComponent"
  style={
    Object {
      "transform": Array [
        Object {
          "rotate": "0deg",
        },
      ],
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 50,
        },
        Object {
          "color": "#333",
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î—•
  </Text>
</View>
`;
